docs/proposals/proposal-002-run.md

```diff
@@ -0,0 +1,326 @@
+
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include manifests or workflow examples
+(though not always required) or even code snippets. If there's any ambiguity
+about HOW your proposal will be implemented, this is the place to discuss them.
+-->
+
+The Green Reviews automated pipeline relies on putting together different reusable GitHub Action workflows to modularise the different moving parts. A workflow runs one or more jobs, and each job runs one or more actions. It may be helpful to familiarise oneself with the documentation on [GitHub Action workflows](https://docs.github.com/en/actions/using-workflows/about-workflows) and especially [Reusing workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) first to better understand the rest of the proposal as it explains some of these concepts well.  The section on [Calling reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow) describes an important concept that will be referenced later in this proposal.
+
+### Definitions
+
+There are different components defined here and shown in the following diagram.
+
+![Green Reviews pipeline components](diagrams/green-reviews-pipeline-components.png "Green Reviews pipeline components")
+
+Let's recap some of the components defined in [Proposal 1](proposal-001-trigger-and-deploy.md):
+1. **Green Reviews pipeline**: the Continuous Integration pipeline which deploys a CNCF project to a test cluster, runs a set of benchmarks while measuring carbon emissions and stores the results. It is implemented by the workflows listed below.
+2. **Cron workflow**: This refers to the initial GitHub Action workflow (described in proposal 1) and which dispatches a project workflow (see next definition), as well as a delete workflow to clean up the resources created by the project workflow.
+3. **Project workflow**: The project workflow is dispatched by the Cron workflow. A project workflow can be, for example, a Falco workflow. A project workflow deploys the project and calls the test workflow (see below). A project workflow can be dispatched more than once if there are multiple project variants/setups. In addition, a Project workflow, which is also just another GitHub Action workflow, contains a list of GitHub Action Jobs. These are a list of test jobs - more info below.
+4. **Delete/cleanup workflow**: This is the one to make sure that the resources created by the project workflow are deleted so the environments go back to the initial state.
+
+This proposal adds the following components:
```

Comment:
I read again the all document and I have a comment on this one. 

There is a couple of issues that are not captured after we took some decisions last week.

I would simplify a bit by saying that:

A project workflow is composed of 3 steps: 
- Create resources
- Benchmark Workflow
- Delete Resources

Let us introduce two new components (with respect to the 4 above):
- Benchmark worfklow: a list of benchmark jobs that needs to be run (in parallel)
- Benchmark Job:  a list of test instructions that are executed on the cluster

A Benchmark workflow has a  1:many relationship with benchmark Jobs.

A Benchmark Job can be:
- internal: defined within the GRT repo
- external: defined outside

So I would so some renaming:
- test workflow -> benchmark worfklow
- test job -> benchmark job

I personally think that benchmark is better then test since it captures more what we are doing. 

Also attaching a picture to give a visual idea:

![Screenshot 2024-07-23 at 18 21 11](https://github.com/user-attachments/assets/5bf3cf96-ffb9-4349-8c45-04a8a400a0e5)

I think this naming is also bit more coherent with the first image. The downside is that we might need more changes